visualization/include/pcl/visualization/pcl_visualizer.h

```diff
@@ -599,6 +599,25 @@ namespace pcl
                    double r = 1.0, double g = 1.0, double b = 1.0,
                    const std::string &id = "", int viewport = 0);
 
+        /** \brief Add a 3d text to the scene
+          * \param[in] text the text to add
+          * \param[in] position the world position where the text should be added
+          * \param[in] orientation the angles of rotation regarding x-, y-, z-axis for the text
```

Comment:
What rotation representation is this exactly? A rodrigues vector? Euler? It's important for that to be clear in the comment.

Reply:
I am currently trying to figure it out from the [VTK documentation](https://www.vtk.org/doc/nightly/html/classvtkProp3D.html#a1c44f66f6ce311d9f38b9e1223f9cee5) and from my experiments but I completly agree, it's not so clear.

Reply:
Don't forget to update this description here with the latest findings, i.e. a ZXY intrinsic rotation. I'm considering even to include a link to wikipedia just to be sure.

That being said, I don't really understand why would VTK would adopt a rotation representation which suffers from gimbal lock. I also wouldn't know how to construct this ZYX rotation from a Rodrigues vector for instance.

Reply:
I'll update the documentation soon with the wikipedia link which seems to be essential!

I'm like you about the VTK choice. I learned recently about the gimbal lock issue with Euler angle. I still don't find a canonical way to handle them, but it's the VTK API. I hesitated to send a mail to VTK, but it doesn't worth it for me. (I don't know the Rodrigues vector, but it's surely more pleasant.)